File: 1.0/src/cz/mg/vulkan/PFNvkUnregisterObjectsNVX.java
package cz.mg.vulkan;

public class PFNvkUnregisterObjectsNVX extends VkFunctionPointer {
    public PFNvkUnregisterObjectsNVX() {
    }

    protected PFNvkUnregisterObjectsNVX(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkUnregisterObjectsNVX(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkUnregisterObjectsNVX(long value) {
        setValue(value);
    }

    public PFNvkUnregisterObjectsNVX(VkInstance instance) {
        super(instance, new VkString("vkUnregisterObjectsNVX"));
    }

    public int call(VkDevice device, VkObjectTableNVX objectTable, int objectCount, VkObjectEntryTypeNVX pObjectEntryTypes, VkUInt32 pObjectIndices){
        return callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), objectTable != null ? objectTable.getVkAddress() : VkPointer.getNullAddressNative(), objectCount, pObjectEntryTypes != null ? pObjectEntryTypes.getVkAddress() : VkPointer.NULL, pObjectIndices != null ? pObjectIndices.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callNative(long vkaddress, long device, long objectTable, int objectCount, long pObjectEntryTypes, long pObjectIndices);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkUnregisterObjectsNVX.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkUnregisterObjectsNVX_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong objectTable, jlong objectCount, jlong pObjectEntryTypes, jlong pObjectIndices, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkUnregisterObjectsNVX f = (PFN_vkUnregisterObjectsNVX)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkObjectTableNVX*)jniLongToPointer(objectTable)),
        *((uint32_t*)jniLongToPointer(objectCount)),
        ((VkObjectEntryTypeNVX*)jniLongToPointer(pObjectEntryTypes)),
        ((uint32_t*)jniLongToPointer(pObjectIndices))
    );
}


jint Java_cz_mg_vulkan_PFNvkUnregisterObjectsNVX_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong objectTable, jint objectCount, jlong pObjectEntryTypes, jlong pObjectIndices){
    (void)env;
    (void)jc;
    PFN_vkUnregisterObjectsNVX f = (PFN_vkUnregisterObjectsNVX)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkObjectTableNVX*)jniLongToPointer(objectTable)),
        (uint32_t)objectCount,
        ((VkObjectEntryTypeNVX*)jniLongToPointer(pObjectEntryTypes)),
        ((uint32_t*)jniLongToPointer(pObjectIndices))
    );
}

